src/main/scala/org/virtuslab/internal/load/reader/ReaderCtx.scala

```diff
@@ -61,7 +58,17 @@ case class ReaderCtx(
       case Some(ReaderState.Mapping(i)) if i == indent => true
       case _                                           => false
 
-  private def closeOpenedScopes(): List[Token] =
+  def isAllowedSpecialCharacter(char: Char): Boolean =
+    stateStack.peek() match
+      case Some(ReaderState.FlowMapping) if char == '}' => false
+      case _                                            => true
```

Comment:
Could you explain why `FlowMapping` + `}` is illegal?

Reply:
From yaml specifications:

>  Flow mappings are denoted by surrounding “{” and “}” characters.

so, It's not allowed, that scalar can contain special characters '}'.  The special indicator '}' is reserved for FlowMapping.